jiant/proj/simple/preprocessing.py

```diff
@@ -22,11 +22,23 @@ def __call__(self, datum):
 
 
 def smart_truncate(dataset: torch_utils.ListDataset, max_seq_length: int, verbose: bool = False):
+    """Truncate data to the length of the longest example in the dataset.
+
+    Args:
+        dataset (torch_utils.ListDataset): ListDataset to truncate if possible.
+        max_seq_length (int): The maximum total input sequence length.
+        verbose (bool): If True, display progress bar tracking truncation progress.
+
+    Returns:
+        Tuple[torch_utils.ListDataset, int]: truncated dataset, and length of the longest sequence.
+
+    """
     if "input_mask" not in dataset.data[0]["data_row"].get_fields():
         raise RuntimeError("Smart truncate not supported")
     valid_length_ls = []
     range_idx = np.arange(max_seq_length)
     for datum in dataset.data:
+        # TODO: document why reshape and max happen here (for cola this isn't necessary).
```

Comment:
This is a hack to cover two kinds of inputs. 

* Regular inputs:`[batch_size, seq_len]`
* Multiple-choice inputs:`[batch_size, num_choices, seq_len]`

Reshape collapses all but the last dimension so you can take max over it. I looks like taking `.max(0)` might make more intuitive sense.